migration/migrator.go

```diff
@@ -0,0 +1,337 @@
+package migration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/eth/tracers"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/ethereum/go-ethereum/trie/trienode"
+	"github.com/ethereum/go-ethereum/trie/zk"
+)
+
+var (
+	tracerType = "prestateTracer"
+	// BedrockTransitionBlockExtraData represents the extradata
+	// set in the very first bedrock block. This value must be
+	// less than 32 bytes long or it will create an invalid block.
+	BedrockTransitionBlockExtraData = []byte("BEDROCK")
+)
+
+type ethBackend interface {
+	ChainDb() ethdb.Database
+	BlockChain() *core.BlockChain
+}
+
+type StateMigrator struct {
+	backend       ethBackend
+	db            ethdb.Database
+	zktdb         *trie.Database
+	mptdb         *trie.Database
```

Comment:
`mptdb` is just a wrapper that implements the logic for reading and writing values from the disk db.
All values are simply stored on disk as hashed key pairs. To read these values, you need a backend like `mptdb` to read them. If we use `mptdb` to store any state after Kroma MPT time, `mptdb` will create a hashed key and store it, and when we read the value, it will get the value with hashed key.

And when you make a request to get historical state, geth does not fetch the state from the owning database, returns the result of the Historical RPC directly.